sphinx/util/inventory.py

```diff
@@ -172,7 +243,16 @@ def escape(string: str) -> str:
                         uri += '#' + anchor
                     if dispname == name:
                         dispname = '-'
-                    entry = ('%s %s:%s %s %s %s\n' %
-                             (name, domainname, typ, prio, uri, dispname))
+
+                    # For names with spaces, we need to know exactly where
+                    # the ref-type string starts. Technically, we should not
+                    # have ':' inside domain or role names, but extensions
+                    # may have some weird role names and they could handle
+                    # them internally to be docutils compatible. As such,
+                    # we encode the length of the name after the priority.
+                    slen = f':{len(name)}' if ' ' in name else ''
+                    entry = '%s%s %s %s:%s %s %s\n' % (
```

Comment:
One alternative that I had in mind is to use a special character for the end of the name, but this means that we assume that this character is not used in the name (e.g., `\x00`). The size of the inventory won't be much bigger but we need to check that the name does not contain such character and raise an exception if this is the case during the writing phase.